src/rest_vol_file.c

```diff
@@ -755,3 +804,31 @@ RV_file_close(void *file, hid_t dxpl_id, void **req)
 
     return ret_value;
 } /* end RV_file_close() */
+
+/*-------------------------------------------------------------------------
+ * Function:    RV_iterate_copy_hid_cb
+ *
+ * Purpose:     Callback for H5Iiterate() that copies the hid_t
+ *              of each library object into a list.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              August, 2023
+ */
+herr_t
+RV_iterate_copy_hid_cb(hid_t obj_id, void *udata)
+{
+    herr_t          ret_value = SUCCEED;
+    object_id_list *id_list   = (object_id_list *)udata;
+
+    if (id_list->obj_count >= id_list->max_obj_count)
```

Comment:
This function shouldn't fail when `obj_count >= max_obj_count`. Rather, the callback should check that `obj_count < max_obj_count` and if so, do

    id_list->obj_id_list[id_list->obj_count] = obj_id;
    id_list->obj_count++;

Otherwise, if `obj_count >= max_obj_count` the callback should just return `H5_ITER_STOP` for short-circuit success. To continue, the function should return `H5_ITER_CONT` instead of `SUCCEED`, but that's just a minor detail since they have the same value anyway. You can look at https://github.com/HDFGroup/vol-daos/blob/master/src/daos_vol_file.c#L3928-L3962 for how the DAOS VOL implements it, though you can probably skip the part about checking the connector name.